uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlementTest.php

```diff
@@ -606,6 +607,91 @@ public function testCanSaveALeavePeriodEntitlementWithAComment() {
     $this->unregisterCurrentLoggedInContactFromSession();
   }
 
+  public function testSaveFromCalculationLogsChangesForAnAlreadyExistingPeriodEntitlement() {
+    $userId = 3;
+    $this->registerCurrentLoggedInContactInSession($userId);
+    $type = new AbsenceType();
+    $type->id = 1;
+    $period = new AbsencePeriod();
+    $period->id = 1;
+    $contact = ['id' => 2];
+
+    $periodEntitlement1 = LeavePeriodEntitlementFabricator::fabricate([
+      'contact_id' => $contact['id'],
+      'period_id' => $period->id,
+      'type_id' => $type->id,
+      'comment' => 'This is a sample comment'
+    ]);
+
+    $entitlementBalance = 3;
+    $this->createLeaveBalanceChange($periodEntitlement1->id, $entitlementBalance);
+
+    $broughtForward = 1;
+    $proRata = 10;
+    $calculation = $this->getEntitlementCalculationMock(
+      $period,
+      $contact,
+      $type,
+      $broughtForward,
+      $proRata
+    );
+
+    $comment = 'Lorem ipsum dolor sit amet...';
+    LeavePeriodEntitlement::saveFromCalculation($calculation, null, $comment);
+
+    //This should create a record in the entitlement log table
+    $leavePeriodEntitlementLog = new LeavePeriodEntitlementLog();
+    $leavePeriodEntitlementLog->entitlement_id = $periodEntitlement1->id;
+    $leavePeriodEntitlementLog->find();
+
+    while($leavePeriodEntitlementLog->fetch()){
+      $entitlementLogs[] = $leavePeriodEntitlementLog;
+    }
+
+    $this->assertCount(1, $entitlementLogs);
+    $leavePeriodEntitlementLog = $entitlementLogs[0];
+
+    $this->assertEquals($entitlementBalance, $leavePeriodEntitlementLog->entitlement_amount);
+    $this->assertEquals($userId, $leavePeriodEntitlementLog->editor_id);
+    $this->assertEquals($periodEntitlement1->comment, $leavePeriodEntitlementLog->comment);
+    $this->assertEquals(new DateTime($periodEntitlement1->created_date), new DateTime($leavePeriodEntitlementLog->created_date));
+  }
+
+  public function testSaveFromCalculationWillNotLogChangesForAFreshlyCreatedPeriodEntitlement() {
+    $userId = 3;
+    $this->registerCurrentLoggedInContactInSession($userId);
+    $type = new AbsenceType();
+    $type->id = 1;
+    $period = new AbsencePeriod();
+    $period->id = 1;
+    $contact = ['id' => 2];
+
+    $broughtForward = 1;
+    $proRata = 10;
+    $calculation = $this->getEntitlementCalculationMock(
+      $period,
+      $contact,
+      $type,
+      $broughtForward,
+      $proRata
+    );
+
```

Comment:
Perhaps in this test, it would be good to assert first that there is not Leave Period Entitlement for this contact and absence type during this period